tests/gireg.lua

```diff
@@ -154,18 +174,28 @@ function gireg.type_int64()
 -- is always 'double', and conversion between double and int64 big
 -- constants is always lossy.  Not sure if it can be solved somehow.
 
---   checkv(R.test_int64(0x7fffffffffffffff), 0x7fffffffffffffff, 'number')
---   checkv(R.test_int64(-0x8000000000000000), -0x8000000000000000, 'number')
---   check(not pcall(R.test_int64, 0x8000000000000000))
---   check(not pcall(R.test_int64, -0x8000000000000001))
+-- UPDATE: It IS working in Lua 5.3 because of its integer support,
+-- so tests are enabled there.
+
+   if nativeIntegers then
+      checkv(R.test_int64(0x7fffffffffffffff), 0x7fffffffffffffff, 'number')
+      checkv(R.test_int64(-0x8000000000000000), -0x8000000000000000, 'number')
```

Comment:
Can this even work correctly? The constant before `-` is applied is larger than the maximum value a signed 64 bit integer can hold, so this should overflow. I doubt that Lua guarantees anything in the presence of integer overflows...